### PR TITLE
US83 - Implement logout functionality

### PR DESCRIPTION
**Pull request by lgonczar**
Also, refactor some code and implemented some missing unit tests

**Merge only with https://github.com/LandRegistry/digital-register-feeder/pull/138**

### CODE REVIEW

File: service/server.py [hunk 1]
```diff
@@ -276,6 +300,59 @@ def get_property_address_index_polygon(geometry_data):
...
+        username=current_user.get_id(),
+        )
+
+
+def _get_search_by_title_number_response(search_term, page_number):
+    title_ref = search_term
+    title = get_register_title(title_ref)
+    if title:
+        # If the title exists store it in the session
+        # TODO: stop storing title in session
```
Comment (camillebaldock) [inline service/server.py]: shall we stop storing them instead of putting a todo? don't think this is used anywhere

File: tests/test_app.py [hunk 2]
```diff
@@ -225,13 +240,99 @@ def test_address_search_with_page_calls_api_correctly(self, mock_get):
...
+        search_term = 'PL9 8TB'
+        self.app.get('/title-search/{}'.format(search_term))
+
+        assert len(mock_get.mock_calls) == 1
+        actual_call = mock_get.mock_calls[0]
+        url_param = actual_call[1][0]
+        assert url_param.endswith('title_search_postcode/{}'.format(search_term))
+
+
+class TestAuthentication:
```
Comment (camillebaldock) [inline tests/test_app.py]: can we split this class in an extra file? really like the extra testing and nicer naming here :+1:

Comment (lgonczar) [inline tests/test_app.py]: It's actually not a bad idea. Will do.

File: service/server.py (hunk 1 above)
Comment (lgonczar) [inline service/server.py]: It is used - the title details page assumes it's there and returns 404 if it's not. I created a story for this.

File: tests/test_app.py (hunk 2 above)
Comment (lgonczar) [inline tests/test_app.py]: I'm wrong! If it's not there, it tries to get it from the API. Can change it now :)

File: service/server.py (hunk 1 above)
Comment (lgonczar) [inline service/server.py]: I was wrong! If it's not there, it tries to get it from the API. Can change it now :)

**Pull request merged**